File: app/models/api/board.rb
module Api
  class Board
    def initialize(board:)
      @board = board
      @last_entry = board.last_entry
    end

    def attributes
      {
        id: board.id,
        title: board.title,
        description: board.description,
        feedIds: board.feeds.ids,
        lastEntry: last_entry_attirbutes
      }
    end

    private

    attr_reader :board, :last_entry

    def last_entry_attirbutes
      last_entry ? Entry.new(entry: last_entry).attributes : nil
    end
  end
end

File: spec/javascript/mocks/Board.ts
import { Board } from "@js/types/types";
import { buildEntry } from "./Entry";

export const buildBoard = (id: number): Board => {
  return {
    id: id,
    title: `board title ${id}`,
    description: `board description ${id}`,
    feedIds: [id],
    lastEntry: buildEntry(1),
  };
};
